pkg/trait/builder.go

```diff
@@ -101,7 +102,14 @@ func (t *builderTrait) Apply(e *Environment) error {
 		}})
 
 	case v1.IntegrationPlatformBuildPublishStrategyBuildah:
+		var platform string
+		var found bool
+		if platform, found = e.Platform.Status.Build.PublishStrategyOptions[builder.BuildahPlatform]; !found {
+			platform = platforms.DefaultSpec().OS + "/" + platforms.DefaultSpec().Architecture + "/" + platforms.DefaultSpec().Variant
```

Comment:
We're still forcing the build within the architecture where the Camel K operator Pod Node is running. Since we now have the platform as a parameter we should let the user manually provide that information without involving any `platforms` dependency.

Reply:
Yeah, if the PublishStrategyOptions haven't this value we proceed with default value retrieved from Node architecture.
Buildah as default use the architecture of Node if you don't specify platform attribute

Reply:
My main concern is we introduce a dependency (with all its maintenance burden) just to get some OS info. Could we try instead using something already available in the Go language (ie, https://pkg.go.dev/runtime#GOOS).

Reply:
I propose another solution... we can invert the logic in this way.

The platform attribute will be initialized and mapped inside buildah if exists, otherwise platform will be ignored